app/src/org/commcare/views/EntityView.java

```diff
@@ -167,7 +167,7 @@ public void setSearchTerms(String[] terms) {
     }
 
     public void refreshViewsForNewEntity(Entity e, boolean currentlySelected, long rowId) {
-        for (int i = 0; i < e.getNumFields(); ++i) {
+        for (int i = 0; i < views.length; ++i) {

```

Comment:
Hey, a little confused what this change represents. Seems like if `e.getNumFields()` was incorrect, it should have been causing buggy behavior. What exactly is the distinction between the 2 expressions?